### PR TITLE
fix(cart): in-memory api does not update collection of carts as expected

### PR DESCRIPTION
**Pull request by lderrickable**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
The cart in memory implementation for the demo app does not accept "cart-item" requests. Additionally, the way we currently update the in-memory database doesn't work. It just silently fails.

## What is the new behavior?
The cart feature in the demo app works as expected, and the cart in-memory services work now.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: libs/cart/driver/in-memory/src/backend/cart-items/cart-items.service.ts [hunk 1]
```diff
@@ -85,40 +85,61 @@ export class DaffInMemoryBackendCartItemsService implements DaffInMemoryDataServ
...
+				items: [
+					...cart.items,
+					this.transformItemInput(itemInput)
+				]
+			}
 		}
-		cart.items = Object.assign([], cart.items);
 
-    return cart;
+    return reqInfo.collection[0];
```
Comment (griest024) [inline libs/cart/driver/in-memory/src/backend/cart-items/cart-items.service.ts:131]: Shouldn't this be `reqInfo.collection[cartIndex]`?

**Review by griest024 — CHANGES_REQUESTED** [review]

File: libs/cart/driver/in-memory/src/backend/cart-items/cart-items.service.ts (hunk 1 above)
Comment (lderrickable) [reply libs/cart/driver/in-memory/src/backend/cart-items/cart-items.service.ts]: definitely. changed

File: apps/demo/src/app/drivers/in-memory/backend/backend.service.ts [hunk 2]
```diff
@@ -33,8 +33,8 @@ export class DemoInMemoryBackendService implements InMemoryDbService {
   }
 
   post(reqInfo: any) {
-    const collectionName = reqInfo.collectionName;
-    if (collectionName === 'cart') {
+		const collectionName = reqInfo.collectionName;
+    if (collectionName.includes('cart')) {
```
Comment (griest024) [inline apps/demo/src/app/drivers/in-memory/backend/backend.service.ts:37]: The list of collection names that cart accepts is provided as a static field so you can do this check like `DaffInMemoryBackendCartRootService.COLLECTION_NAMES.indexOf(collectionName) > -1`. Its more verbose but less brittle.

**Review by griest024 — CHANGES_REQUESTED** [review]

File: apps/demo/src/app/drivers/in-memory/backend/backend.service.ts (hunk 2 above)
Comment (lderrickable) [reply apps/demo/src/app/drivers/in-memory/backend/backend.service.ts]: changed

**Review by griest024 — APPROVED** [review]

**Review by griest024 — APPROVED** [review]

**Pull request merged**